manim/utils/hashing.py

```diff
@@ -0,0 +1,150 @@
+
+        Returns
+        -------
+        `any`
+            Python object that JSON encoder will recognize
+
+        """
+        if inspect.isfunction(obj) and not isinstance(obj, ModuleType):
+            r = inspect.getclosurevars(obj)
+            x = {**copy.copy(r.globals), **copy.copy(r.nonlocals)}
+            for i in list(x):
+                # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
+                if isinstance(x[i], ModuleType):
+                    del x[i]
+            return {'code': inspect.getsource(obj),
+                    'nonlocals': x}
+        elif isinstance(obj, np.ndarray):
+            return list(obj)
+        elif hasattr(obj, "__dict__"):
+            return getattr(obj, '__dict__')
+        elif isinstance(obj, np.uint8):
+            return int(obj)
+        try:
+            return json.JSONEncoder.default(self, obj)
+        except TypeError:
+            # This is used when the user enters an unknown type in CONFIG. Rather than throwing an error, we transform
+            # it into a string "Unsupported type for hashing" so it won't affect the hash.
+            return "Unsupported type for hashing"
+
+
+def get_json(object):
+    """Recursively serialize object to JSON. Use CustomEncoder class above.
+
+    Paramaters
+    ----------
+    dict_config : :class:`dict`
+        The dict to flatten
+
+    Returns
+    -------
+    `str` 
+        the object flattened
+    """
+    return json.dumps(object, cls=CustomEncoder)
+
+
+def get_camera_dict_for_hashing(camera_object):
+    """Remove some keys from cameraobject.__dict__ that are useless for the caching functionnality and very heavy. 
+
+    Parameters
+    ----------
+    object_camera : :class:``~.Camera`
+        The camera object used in the scene
+
+    Returns
+    -------
+    `dict`
+        Camera.__dict__ but cleaned.
+    """
+    camera_object_dict = copy.copy(camera_object.__dict__)
+    # We have to clean a little bit camera_dict, as pixel_array and background are two very big numpy array.
+    # They are not essential to caching process.
+    # We also have to remove pixel_array_to_cairo_context as it conntains uses memory adress (set randomly). See l.516 get_cached_cairo_context in camera.py
+    for to_clean in ['background', 'pixel_array', 'pixel_array_to_cairo_context']:
+        camera_object_dict.pop(to_clean, None)
+    return camera_object_dict
+
+
+def get_hash_from_play_call(camera_object, animations_list, current_mobjects_list):
+    """Take the list of animations and a list of mobjects and output their hash. Is meant to be used for `scene.play` function.
+
+    Parameters
+    -----------
+    object_camera : :class:``~.Camera`
+        The camera object used in the scene
+
+    animations_list : :class:`list`
+        The list of animations 
+
+    current_mobjects_list : :class:`list`
+        The list of mobjects.
```

Comment:
Inconsistent usage of periods

```suggestion
        The camera object used in the scene.

    animations_list : :class:`list`
        The list of animations.

    current_mobjects_list : :class:`list`
        The list of mobjects.
```